### PR TITLE
Added VT-40 keyboard

### PR DESCRIPTION
**Pull request by guidoism**
## Description

Added VT-40 keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ x] Keyboard (addition or update)
- [ ] Keymap/layout (addition or update)
- [ ] Documentation

## Checklist

- [ x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/vt40/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [0] = { ENCODER_CCW_CW(KC_PGUP, KC_PGDN), ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+    [1] = { ENCODER_CCW_CW(KC_PGUP, KC_PGDN), ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+    [2] = { ENCODER_CCW_CW(KC_PGUP, KC_PGDN), ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+    [3] = { ENCODER_CCW_CW(KC_PGUP, KC_PGDN), ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+    [4] = { ENCODER_CCW_CW(KC_PGUP, KC_PGDN), ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+    [5] = { ENCODER_CCW_CW(KC_PGUP, KC_PGDN), ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+    [6] = { ENCODER_CCW_CW(KC_PGUP, KC_PGDN), ENCODER_CCW_CW(KC_VOLD, KC_VOLU) }
```
Comment (waffle87) [inline keyboards/vt40/keymaps/default/keymap.c:272]:
```suggestion
    [1] = { ENCODER_CCW_CW(_______, _______), ENCODER_CCW_CW(_______, _______) },
    [2] = { ENCODER_CCW_CW(_______, _______), ENCODER_CCW_CW(_______, _______) },
    [3] = { ENCODER_CCW_CW(_______, _______), ENCODER_CCW_CW(_______, _______) },
    [4] = { ENCODER_CCW_CW(_______, _______), ENCODER_CCW_CW(_______, _______) },
    [5] = { ENCODER_CCW_CW(_______, _______), ENCODER_CCW_CW(_______, _______) },
    [6] = { ENCODER_CCW_CW(_______, _______), ENCODER_CCW_CW(_______, _______) }
```

**Review by waffle87 — CHANGES_REQUESTED** [review]
I suggest you tidy up the keymap and remove the code that does not apply to this keyboard (and was copy and pasted from the default Planck keymap) like audio and midi keycodes, other audio code, etc.

File: keyboards/vt40/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+  _ADJUST
+};
+
+enum planck_keycodes {
+  PLOVER = SAFE_RANGE,
+  LOWER,
+  RAISE,
+  BACKLIT,
+  EXT_PLV
+};
```
[the remark this answers is not in the document]
Comment (guidoism) [reply keyboards/vt40/keymaps/default/keymap.c:23]: Copied over.

[the remark this answers is not in the document]
Comment (waffle87) [reply keyboards/vt40/keymaps/default/keymap.c:23]: As per the [PR Checklist](https://docs.qmk.fm/pr_checklist):
> - default keymaps should be "pristine"
    - bare minimum to be used as a "clean slate" for another user to develop their own user-specific keymap

They should be removed, especially if they probably don't serve any purpose to you.

[the remark this answers is not in the document]
Comment (guidoism) [reply keyboards/vt40/keymaps/default/keymap.c]: Yeah, got it. I was a little to quick to put this in. Sorry.

File: keyboards/vt40/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+   * |      | Reset|      |      |      |      |      |      |      |      |      |  Del |
+   * |------+------+------+------+------+-------------+------+------+------+------+------|
+   * |      |      |      |      |      |AGnorm|AGswap|Qwerty|Colemk|Dvorak|      |      |
+   * |------+------+------+------+------+------|------+------+------+------+------+------|
+   * |      |      |      |      |      |             |      |      |      |      |      |
+   * |------+------+------+------+------+------+------+------+------+------+------+------|
+   * |      |      |      |      |      |             |      |      |      |      |      |
+   * `-----------------------------------------------------------------------------------'
+   */
+  [_ADJUST] = LAYOUT_ortho_4x12(
```
[the remark this answers is not in the document]
Comment (waffle87) [reply keyboards/vt40/keymaps/default/keymap.c:126]: Looks like you added a definition for the `ADJUST` keycode, and added it to the ASCII layout representation you have, but not actually mapped in the keymap.

[the remark this answers is not in the document]
Comment (guidoism) [reply keyboards/vt40/keymaps/default/keymap.c:126]: Oh my god I need a nap. 🫤

**Review by waffle87 — APPROVED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/vt40/keyboard.json [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        "command": false,
+        "console": false,
+        "encoder": true,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
+    },
+    "matrix_pins": {
+        "cols": ["GP6", "GP7", "GP8", "GP9", "GP10", "GP11", "GP12",
+                 "GP13", "GP21", "GP20", "GP19", "GP18", "GP22"],
```
Comment (dunk2k) [inline keyboards/vt40/keyboard.json:18]:
```suggestion
        "cols": ["GP6", "GP7", "GP8", "GP9", "GP10", "GP11", "GP12", "GP13", "GP21", "GP20", "GP19", "GP18", "GP22"],
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Pull request merged**